CMakeLists.txt

```
@@ -0,0 +1,131 @@
+  check_fortran_compiler_flag("-fallow-invalid-boz" _boz_flag)
+  check_fortran_compiler_flag("-fallow-argument-mismatch" _argmis_flag)
+  if(_boz_flag)
+    set(CMAKE_Fortran_FLAGS               "${CMAKE_Fortran_FLAGS} -fallow-invalid-boz" )
+  endif()
+  if(_argmis_flag)
+    set(CMAKE_Fortran_FLAGS               "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch" )
+  endif()
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
+  set(CMAKE_Fortran_FLAGS_RELWITHDEBINFO "-g")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -g -W -fbounds-check")
+
+elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
+
+  # Copied from MOM5/bin/mkmf.template.nci
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-alias -safe-cray-ptr -fpe0 -ftz -assume byterecl -i4 -r8 -traceback -nowarn -check noarg_temp_created -assume nobuffered_io -convert big_endian -grecord-gcc-switches -fp-model precise -fp-model source -align all")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-g3 -O2 -xCORE-AVX2 -debug all -check none")
+  set(CMAKE_Fortran_FLAGS_RELWITHDEBINFO "-g3 -O2 -xCORE-AVX2 -debug all -check none")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-g3 -O0 -debug all -check -check noarg_temp_created -check nopointer -warn -warn noerrors -ftrapuv")
+
+elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "IntelLLVM")
+
+  # Copied from MOM5/bin/mkmf.template.nci
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-alias -safe-cray-ptr -fpe0 -ftz -assume byterecl -i4 -r8 -traceback -nowarn -check noarg_temp_created -assume nobuffered_io -convert big_endian -grecord-gcc-switches -fp-model precise -fp-model source -align all")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-g3 -O2 -xCORE-AVX2 -debug all -check none")
+  set(CMAKE_Fortran_FLAGS_RELWITHDEBINFO "-g3 -O2 -xCORE-AVX2 -debug all -check none")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-g3 -O0 -debug all -check -check noarg_temp_created -check nopointer -warn -warn noerrors -ftrapuv")
+
+else()
+  message(WARNING "Fortran compiler with ID ${CMAKE_Fortran_COMPILER_ID} will be used with CMake default options")
+endif()
+
+################################################################################
+
+find_package(MPI REQUIRED COMPONENTS Fortran)
+find_package(fms COMPONENTS R8 REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(MOCSY REQUIRED IMPORTED_TARGET "mocsy")
+
+add_library(gtracers STATIC)
+
+target_sources(gtracers PRIVATE
+  generic_tracers/FMS_coupler_util.F90
+  generic_tracers/FMS_ocmip2_co2calc.F90
+  generic_tracers/generic_abiotic.F90
+  generic_tracers/generic_age.F90
+  generic_tracers/generic_argon.F90
+  generic_tracers/generic_BLING.F90
+  generic_tracers/generic_blres.F90
+  generic_tracers/generic_CFC.F90
+  generic_tracers/generic_COBALT.F90
+  generic_tracers/generic_ERGOM.F90
+  generic_tracers/generic_miniBLING.F90
+  generic_tracers/generic_SF6.F90
+  generic_tracers/generic_TOPAZ.F90
+  generic_tracers/generic_tracer.F90
+  generic_tracers/generic_tracer_utils.F90
+  generic_tracers/generic_WOMBATlite.F90
```

WOMBATmid has since been added

```suggestion
  generic_tracers/generic_WOMBATlite.F90
  generic_tracers/generic_WOMBATmid.F90
```

---

Done in [0d984a2](https://github.com/ACCESS-NRI/GFDL-generic-tracers/pull/17/commits/0d984a2ef67a36737a574176cc3ce308ad8e7305)